lib/github-pages-health-check/domain.rb

```diff
@@ -283,7 +283,7 @@ def pages_dot_github_dot_com?
 
       # Is this domain owned by GitHub?
       def github_domain?
-        !!host.downcase.end_with?("github.com")
+        !!host.match(/(\A|\.)github\.com\.?\z/i)
```

Comment:
Without a test, it's a bit difficult to know what potential problematic input this also handles. In general regular expressions are much slower in Ruby than string comparisons, but moreover they're a bit harder to audit/grok. I see it matches more cases like `github.com.` (trailing period) and enforces either strictly github.com or a subdomain thereof, but it's not necessarily clear what other cases you might want to include. 

Checking `host.downcase.eql?("github.com") || host.downcase.end_with?(".github.com")` could be a more direct alternative if you're interested.

Reply:
👋 Always nice to see you around!

Thanks for the feedback 🙇. Here goes: https://github.com/github/pages-health-check/pull/172